### PR TITLE
Improve install role

### PR DESCRIPTION
**Pull request by l00ptr**

### CODE REVIEW

**Comment by l00ptr** [comment]
any feedback about that MR ?

**Comment by coudot** [comment]
Hello @l00ptr 

thanks for the contributions. I do not use the ansible role, I don't have any feedback to give.

For the moment the Molecule test seems to fail, you should fix it before we merge.

**Comment by xavierba** [comment]
This will need to be rebased once #16, which addresses only linting, has been merged.

File: roles/install/tasks/debian.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     mode: 0644
 
 - name: Import LLNG GPG KEY
-  ansible.builint.apt_key:
+  ansible.builtint.apt_key:
```
Comment (xavierba) [inline roles/install/tasks/debian.yml]: there's still a typo here: builtint --> builtin
also, please squash with previous commit

File: roles/install/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+__default_pkgs:
+  - lemonldap-ng
+  - lemonldap-ng-doc
+
+__redhat_apache_pkgs:
+  - httpd
+  - mod_perl
+  - mod_fcgid
```
Comment (xavierba) [inline roles/install/vars/main.yml]: mod_perl and mod_fcgid should be pulled in by the rpm packages dependencies.

File: roles/install/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+__redhat_nginx_pkgs:
+  - nginx
+  - lemonldap-ng-fastcgi-server
+  - lemonldap-ng-nginx
```
Comment (xavierba) [inline roles/install/vars/main.yml]: Starting with llng 2.19.0, -nginx subpackage doesn't exist anymore, the nginx configurations have been move to their respective sub-packages.

File: roles/install/vars/main.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+__redhat_default_pkgs:
+  - perl-LWP-Protocol-https
```
Comment (xavierba) [inline roles/install/vars/main.yml]: This is missing in the RPMs Requires, I will fix that, thanks :-)

Comment (l00ptr) [reply roles/install/vars/main.yml]: so let me know when it's done on the RPM side and i will also remove that package

File: roles/install/vars/main.yml (hunk 2 above)
Comment (l00ptr) [reply roles/install/vars/main.yml]: removed!

File: roles/install/tasks/debian.yml (hunk 1 above)
Comment (l00ptr) [reply roles/install/tasks/debian.yml]: fixed

**Comment by l00ptr** [comment]
@xavierba [Ansible 2.17 do not work properly with python 3.6](https://github.com/ansible/ansible/issues/82068) (which is the default on rockylinux8); we probably need to install python39 (or greater) or officially don't support rockylinux8 with our collection... any idea about that specific topic ?

File: roles/install/vars/main.yml (hunk 4 above)
Comment (xavierba) [reply roles/install/vars/main.yml]: The dependency has been explicitly added with https://gitlab.ow2.org/lemonldap-ng/lemonldap-ng/-/merge_requests/513

But anyway, this was already an indirect dependency from perl-Soap-Lite and perl-Sentry-Raven.

File: roles/install/tasks/set-domain.yml [hunk 5]
```diff
@@ -1,51 +1,55 @@
...
-    - "{{ lemonldap_webserver_conf_search_res.files }}"
-    - "{{ lemonldap_apache_conf_search_res.files }}"
-    - "{{ lemonldap_nginx_conf_search_res.files }}"
-    - "{{ lemonldap_ini_search_rest.files }}"
-    - {'path': '/var/lib/lemonldap-ng/conf/lmConf-1.json'}
-  notify:
-    - Reload Webserver
-    - Enable and reload lemonldap
+- name: Reconfigure lemonldap (based on search)
+  block:
```
Comment (xavierba) [inline roles/install/tasks/set-domain.yml]: there's only one module call under this 'block', so 'block' is useless.

File: roles/install/tasks/set-domain.yml [hunk 6]
```diff
@@ -57,27 +61,29 @@
...
-    - ansible_os_family == "Debian"
-  notify: Reload Webserver
+    - name: Enable apache module required by lemonldap config
+      ansible.builtin.command: a2enmod {{ item }}
+      register: res_a2enmod
+      changed_when: "'already enabled' not in res_a2enmod.stdout"
+      with_items:
+        - headers
+        - rewrite
+      when:
```
Comment (xavierba) [inline roles/install/tasks/set-domain.yml]: 'when' is duplicated and thus can be moved at the 'block' level.

File: roles/install/vars/main.yml (hunk 4 above)
Comment (l00ptr) [reply roles/install/vars/main.yml]: so i removed that part!

**Comment by xavierba** [comment]
Looks good to me, thank you @l00ptr !

**Comment by xavierba** [comment]
Just one thing, could you please squash the various fixup commits with the commit they fix ?
I'll merge the PR once done.

**Comment by l00ptr** [comment]
> Just one thing, could you please squash the various fixup commits with the commit they fix ? I'll merge the PR once done.

!done

File: roles/install/tests/test.yml
Comment (xavierba) [inline roles/install/tests/test.yml:1]: This file deletion probably belongs to " Remove unused test dir on install role" commit

**Review by xavierba — COMMENTED** [review]
Other than the file deletion in an unrelated commit, LGTM

**Pull request merged**